src/DynamoRevit/DynamoRevit.cs

```diff
@@ -445,12 +445,12 @@ private static RevitDynamoModel InitializeCoreModel(DynamoRevitCommandData comma
                 Environment.SpecialFolder.ApplicationData), 
                 "Dynamo", "Dynamo Revit");
             var commonDataFolder = Path.Combine(Environment.GetFolderPath(
-                Environment.SpecialFolder.CommonApplicationData), 
-                "Dynamo", "Dynamo Revit");
+                Environment.SpecialFolder.CommonApplicationData),
+                "Autodesk", "RVT " + DocumentManager.Instance.CurrentUIApplication.Application.VersionNumber, "Dynamo");
```

Comment:
@QilongTang I'm not sure how but is it possible for the instance to be null?

Reply:
@mjkkirschner It is, although a funny workflow. You need to open Revit-> open a document -> open Dynamo -> close current Revit document to get into this case. As long as you have a Revit document open, you will have this reference. Also remember that first time opening Revit, you will not be able to use Dynamo without opening a Revit doc.
![image](https://user-images.githubusercontent.com/3942418/49041177-8db88700-f192-11e8-9745-6c23f0464485.png)

Reply:
@mjkkirschner I think instead of making this always fetched runtime , I will try to record it when Dynamo launches because it is guaranteed that when D4R launches, there is an active Revit doc (at least for Revit with UI). Let me know if there is any objection

Reply:
@QilongTang I feel we may already do that somewhere...

Reply:
That sounds like a regression - it should be able to reattach to a new project... but I think you can definitely get into a weird state with an orphaned dynamo.